storage/src/mmr/proof.rs

```diff
@@ -534,6 +538,7 @@ pub(crate) fn nodes_required_for_multi_proof(
     // Collect all required node positions
     //
     // TODO(#1472): Optimize this loop
+    assert!(!locations.is_empty(), "empty locations");
```

Comment:
I'm assuming there is a "precheck" call that prevents us from hitting this unless called directly (following the error changes).

Reply:
Ah yep, found it.